### PR TITLE
Doc for  boolean function and examples

### PR DESCRIPTION
**Pull request by jtlap**

### CODE REVIEW

File: include/boost/simd/function/compare_equal.hpp [hunk 1]
```diff
@@ -16,28 +16,24 @@ namespace boost { namespace simd
...
     @ingroup group-reduction
-    Function object implementing compare_equal capabilities
-
-    Returns a boolean that is the result of the lexicographic test for equality of all elements of
+    This function object returns a boolean that is the result of the lexicographic test for equality of all elements of
     the entries, i.e. return @ref true if and only if all corresponding entries elements are equal.
 
-    @par Semantic:
+   @par Note
+
```
Comment (jfalcou) [inline include/boost/simd/function/compare_equal.hpp]: there is no such thing as 'lexicographic equality'. Just state above that it returns all(x==y) and get rid of the Note block

File: include/boost/simd/function/if_else.hpp [hunk 2]
```diff
@@ -17,13 +17,10 @@ namespace boost { namespace simd
...
-    This function object returns the If @c c is @ref True returns @c t else returns @c f
-    If vectors, the types involved in the call must share the same number of elements.
-
-   @par Note
-
-      For every parameters @c c of type @c C, @c t and @c f of type @c T : `T r = if_else(c, t, f)` is similar to: `T r = c ? t : f`
+    This function object returns @c t (respectively @c f) if @c c
+    is @ref True (respectively  @ref False)
 
+    If vectors, the types involved in the call must share the same number of elements.
```
Comment (jfalcou) [inline include/boost/simd/function/if_else.hpp]: This is a precondition so @pre If both parameters are pack, they must have same cardinal

File: include/boost/simd/function/if_minus.hpp [hunk 3]
```diff
@@ -16,27 +16,18 @@ namespace boost { namespace simd
...
  /*!
-
     @ingroup group-boolean
-    Function object implementing if_minus capabilities
-
-    The function returns either the second parameter or the difference between the
-    second and third parameter, depending on whether the first parameter is
-	@ref False or @ref True .
+    This function object returns @c x-y (respectively @c x)
+    if @c c is @ref True (respectively  @ref False)
```
Comment (jfalcou) [inline include/boost/simd/function/if_minus.hpp]: why do you cut your line like this, it shoudl on a single one ?

File: include/boost/simd/function/ifnot_dec.hpp [hunk 4]
```diff
@@ -16,31 +16,23 @@ namespace boost { namespace simd
...
 
  /*!
-
     @ingroup group-boolean
-    Function object implementing ifnot_dec capabilities
-
-    Decrements a value by @ref One if a predicate is @ref False.
-
-    @par Semantic:
+    This function object returns  @c x (respectively @c x-1)
```
Comment (jfalcou) [inline include/boost/simd/function/ifnot_dec.hpp]: This comment has to be applid to all conditionnal operation like ifdec etc ...

Just state something like This function objkect performs a conditionnal decrement on x based on the status of the complement of c.

File: include/boost/simd/function/logical_and.hpp [hunk 5]
```diff
@@ -16,33 +16,24 @@ namespace boost { namespace simd
...
 
  /*!
-
     @ingroup group-boolean
-    Function object implementing logical_and capabilities
-
-    return the logical and of the two parameters.
+    This function object returns the logical and of its two parameters.
 
     The operands must share the same @ref cardinal_of value.
```
Comment (jfalcou) [inline include/boost/simd/function/logical_and.hpp]: This is a precondition

File: include/boost/simd/function/logical_and.hpp [hunk 6]
```diff
@@ -16,33 +16,24 @@ namespace boost { namespace simd
...
 
     Infix notation can be used with operator '&&'.
 
-    @warningbox{Take care that using && in scalar mode does short-circuit and does not in simd mode.}
-
-    @par Semantic:
+    @warningbox{Take care that using && in scalar mode does short-circuit
```
Comment (jfalcou) [inline include/boost/simd/function/logical_and.hpp]: Take care that SIMD logical operation do not short-circuit.

File: include/boost/simd/function/negifnot.hpp [hunk 7]
```diff
@@ -16,25 +16,16 @@ namespace boost { namespace simd
 {
```
Comment (jfalcou) [inline include/boost/simd/function/negifnot.hpp:17]: Why is this guy not named if_notneg ?

**Review by jfalcou — CHANGES_REQUESTED** [review]

File: include/boost/simd/function/if_minus.hpp (hunk 3 above)
Comment (jtlap) [reply include/boost/simd/function/if_minus.hpp]: This does not affect the final doc output

File: include/boost/simd/function/negifnot.hpp (hunk 7 above)
Comment (jtlap) [reply include/boost/simd/function/negifnot.hpp:17]: historical and forgotten, I presume... We can change.

Comment (jtlap) [reply include/boost/simd/function/negifnot.hpp:17]: I will change in another PR

**Review by jfalcou — APPROVED** [review]

**Pull request merged**